.github/workflows/sync-test.yml

```
@@ -0,0 +1,128 @@
+---
+concurrency: ci-$ # Only a single workflow can be executed concurrently
+name: Sync Fullnode Test
+on: # yamllint disable-line rule:truthy
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: The environment to run against
+        type: choice
+        options: [testnet]
+  workflow_call:
+    inputs:
+      environment:
+        description: The environment to run against
+        type: string
+        required: true
+    outputs:
+      terraform_output:
+        description: Terraform output
+        value: ${{ jobs.run_test.outputs.terraform_output }}
+      ansible_output:
+        description: Ansible output
+        value: ${{ jobs.run_test.outputs.ansible_output }}
+    secrets:
+      AWS_ROLE_ARN:
+        required: true
+      AWS_S3_BLADE_BUCKET:
+        required: true
+      VAULT_PASSWORD:
+        required: true
+      PERSONAL_ACCESS_USER:
+        required: true
+      PERSONAL_ACCESS_TOKEN:
+        required: true
+
+permissions:
+  id-token: write
+  contents: read
+  security-events: write
+
+jobs:
+  check_network:
+    name: Check if the network is already deployed
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.environment }}
+    outputs:
+      check_output: ${{ steps.check_state_file.outputs.resources }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.1.1
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4.0.1
+        with:
+          aws-region: ${{ vars.AWS_REGION }}
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+      - name: Retrieve state file from s3
+        id: retrieve_state
+        run: echo "retrieve_state_output=$(aws s3 cp s3://${{ secrets.AWS_S3_BLADE_BUCKET }}/${{ vars.TF_STATE_PATH }} state.json)" >> $GITHUB_OUTPUT
+      - name: Check state file
+        id: check_state_file
+        if: contains(steps.retrieve_state.outputs.retrieve_state_output, 'download')
+        run: echo "resources=$(cat state.json | jq -r '.resources' | jq length)" >> $GITHUB_OUTPUT
+  run_test:
+    name: Run the fullnode sync test
+    runs-on: ubuntu-latest
+    needs: check_network
```

```suggestion
    needs: check_network
    if: needs.check_network.outputs.check_output > 0
```

Check if the network is deployed